marc_export/src/main/java/whelk/export/marc/ProfileExport.java

```diff
@@ -357,9 +357,11 @@ private void exportDocument(Document document, ExportProfile profile, MarcRecord
 
         String profileName = profile.getProperty("name", "unknown");
         String systemId = document.getShortId();
-        if (exportedIDs.contains(systemId))
-            return;
-        exportedIDs.add(systemId);
+        synchronized (exportedIDs) {
+            if (exportedIDs.contains(systemId))
+                return;
+            exportedIDs.add(systemId);
+        }
```

Comment:
I suggest keeping the concurrent set and:
```suggestion
        if (!exportedIDs.add(systemId)) {
            return;
        }
```

Reply:
Hm. While also correct (ofc), I like the original one better. I think it's easier to read and understand. Yes, its subjective, I guess.